apps/portals/src/AppInitializer.tsx

```diff
@@ -227,87 +122,28 @@ function AppInitializer(props: { children?: React.ReactNode }) {
 
     window.addEventListener('click', updateSynapseCallbackCookie)
 
-    // Technically, the AppInitializer is only mounted once during the portal app lifecycle.
-    // But it's best practice to clean up the global listener on component unmount.
-
+    // Clean up the global listener on component unmount.
     return () => {
       window.removeEventListener('click', updateSynapseCallbackCookie)
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps -- run only on mount
   }, [])
 
-  useDetectSSOCode({
-    onSignInComplete: () => {
-      redirectAfterSSO()
-    },
-    onTwoFactorAuthRequired: (twoFactorAuthError) => {
-      setTwoFaErrorResponse(twoFactorAuthError)
-      setShowLoginDialog(true)
-    },
-  })
-
-  const onSignInClicked = useCallback(() => {
-    setShowLoginDialog(true)
-  }, [])
-
-  const handleCloseLoginDialog = React.useCallback(() => {
-    setShowLoginDialog(false)
-  }, [])
-
-  const clonedChildren = useMemo(
-    () =>
-      React.Children.map(props.children, (child: any) => {
-        if (!child) {
-          return false
-        } else {
-          const props: SignInProps = {
-            showLoginDialog: showLoginDialog,
-            twoFaErrorResponse: twoFaErrorResponse,
-            getSession: getSession,
-            onSignIn: onSignInClicked,
-            userProfile: userProfile,
-            resetSession: resetSession,
-            handleCloseLoginDialog: handleCloseLoginDialog,
-          }
-          return React.cloneElement(child, props)
-        }
-      }),
-    [
-      getSession,
-      handleCloseLoginDialog,
-      onSignInClicked,
-      props.children,
-      resetSession,
-      showLoginDialog,
-      userProfile,
-    ],
-  )
```

Comment:
We were passing all of this data in by cloning the children and and attaching the data as new props. Now, it's all in context so child components can just access this data if needed.